extensions/wikia/NjordPrototype/NjordController.class.php

```diff
@@ -0,0 +1,325 @@
+			$current = 1;
+			while ( $subpages->valid() ) {
+				$curSub = $subpages->current();
+				if ( $current == $curSub->getSubpageText() ) {
+					$current++;
+				}
+				$subpages->next();
+			}
+			$this->ctitle = self::MAINPAGE_PAGE . "/" . $current;
+		}
+	}
+
+	public function mom() {
+	}
+
+	public function reorder() {
+		$params = $this->getRequest()->getParams();
+
+		$pageTitleObj = Title::newFromText( $params[ 'page' ] );
+		$pageArticleObj = new Article( $pageTitleObj );
+		$content = $pageArticleObj->getContent();
+
+		$content = mb_ereg_replace( '<modula.*/>', '', $content, 'sU' );
+		$moduleTags = [ ];
+		foreach ( $params[ 'left' ] as $raw ) {
+			$data = json_decode( $raw );
+			$moduleTags[ ] = Xml::element( 'modula', $attribs = [
+				'align' => 'left',
+				'title' => $data->text,
+				'content-title' => $data->title,
+			] );
+		}
+		$content = mb_ereg_replace( '(<mainpage-leftcolumn-start.*/>)\s*', "\\1\n" . implode( $moduleTags, "\n" ) . "\n", $content, 'sU' );
+		$moduleTags = [ ];
+		foreach ( $params[ 'right' ] as $raw ) {
+			$data = json_decode( $raw );
+			$moduleTags[ ] = Xml::element( 'modula', $attribs = [
+				'align' => 'right',
+				'title' => $data->text,
+				'content-title' => $data->title,
+			] );
+		}
+		$content = mb_ereg_replace( '(<mainpage-rightcolumn-start.*/>)\s*', "\\1\n" . implode( $moduleTags, "\n" ) . "\n", $content, 'sU' );
+		// save and purge
+		$pageArticleObj->doEdit( $content, '' );
+		$pageArticleObj->doPurge();
+	}
+
+	public function upload() {
+		if ( $this->getRequest()->wasPosted() ) {
+			$url = $this->getRequest()->getVal( 'url', false );
+			try {
+				if ( $url ) {
+					list( $tempPath, $EIA, $errorMessage ) = $this->uploadFromUrl( $url );
+				} else {
+					list( $tempPath, $EIA, $errorMessage ) = $this->uploadFromFile();
+				}
+			} catch ( Exception $exception ) {
+				$EIA = false;
+				$errorMessage = $exception->getMessage();
+			}
+
+			$this->getResponse()->setFormat( 'json' );
+			$this->getResponse()->setVal( 'isOk', $EIA );
+			if ( $EIA === true ) {
+				$stash = RepoGroup::singleton()->getLocalRepo()->getUploadStash();
+				$stashFile = $stash->stashFile( $tempPath );
+				$this->getResponse()->setVal( 'url', wfReplaceImageServer( $stashFile->getThumbUrl( static::THUMBNAILER_SIZE_SUFIX ) ) );
+				$this->getResponse()->setVal( 'filename', $stashFile->getFileKey() );
+			} else {
+				$this->getResponse()->setVal( 'errTitle', wfMessage( 'hero-image-error')->text() );
+				$this->getResponse()->setVal( 'errMessage', $errorMessage );
+			}
+		}
+	}
+
+	public function saveTitleAndDescription() {
+		$description = $this->getRequest()->getVal( 'description', false );
+
+		$success = false;
+		$wikiDataModel = $this->getWikiData();
+
+		if ( $description ) {
+			$wikiDataModel->description = $description;
+		}
+
+		if ( $description ) {
+			$success = $this->setWikiData( $wikiDataModel );
+		}
+
+		$this->getResponse()->setVal( 'success', $success );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "success" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)